domain/tests/src/Unit/DomainValidatorTest.php

```diff
@@ -0,0 +1,126 @@
+use Drupal\domain\DomainValidator;
+
+/**
+ * Tests domain record validation.
+ *
+ * @group domain
+ */
+class DomainValidatorTest extends UnitTestCase {
+
+  /** @var \Drupal\domain\DomainValidator */
+  private $object;
+
+  /** @var \GuzzleHttp\Client */
+  public $client;
+
+  /**
+   * Set up a service container and SUT (Subject Under Test).
+   *
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+
+    /** @var \Drupal\Core\Config\ImmutableConfig $config */
+    $config = $mock = $this->getMockBuilder('Drupal\Core\Config\ImmutableConfig')
+      ->disableOriginalConstructor()->getMock();
+    $mock->expects(static::any())->method('get')
+      ->willReturnMap([
+        ['allow_non_ascii', FALSE],
+        ['www-prefix', FALSE]
+      ]);
+
+    /** @var \Drupal\Core\Config\ConfigFactory $factory */
+    $factory = $mock = $this->getMockBuilder('Drupal\Core\Config\ConfigFactory')
+      ->disableOriginalConstructor()->getMock();
+    $mock->expects(static::once())->method('get')->willReturn($config);
+
+    /** @var \Drupal\Core\Entity\EntityStorageBase $storage */
+    $storage = $mock = $this->getMockBuilder('Drupal\Core\Entity\EntityStorageBase')
+      ->disableOriginalConstructor()->getMock();
+    $mock->expects(static::any())->method('loadByProperties')->willReturn(array());
+
+    /** @var \Drupal\Core\Entity\EntityTypeManager $manager */
+    $manager = $mock = $this->getMockBuilder('Drupal\Core\Entity\EntityTypeManager')
+      ->disableOriginalConstructor()->getMock();
+    $mock->expects(static::any())->method('getStorage')
+      ->willReturn($storage);
+
+    /** @var \Drupal\Core\Extension\ModuleHandler $handler */
+    $handler = $mock = $this->getMockBuilder('Drupal\Core\Extension\ModuleHandler')
+      ->disableOriginalConstructor()->getMock();
+
+    $response = $this->getMockBuilder('GuzzleHttp\Psr7\Response')
+      ->disableOriginalConstructor()->getMock();
+    $response->expects(static::any())->method('getStatusCode')
+      ->willReturn(200);
+
+    $client = $this->getMockBuilder('GuzzleHttp\Client')->setMethods(['request'])
+      ->disableOriginalConstructor()->getMock();
+    $client->expects(static::any())->method('request')
+      ->will(static::returnValue($response));
+
+    $this->object = new DomainValidator($handler, $factory, $client, $manager);
+  }
+
+  /**
+   * @param string $host
+   * @param int $result
+   *
+   * @dataProvider provider
+   */
+  public function testValidateStaticInspection($host, $result) {
+    /** @var \Drupal\domain\Entity\Domain $domain */
+    $domain = $mock = $this->getMockBuilder('Drupal\domain\Entity\Domain')
+      ->disableOriginalConstructor()->getMock();
+    $mock->expects(static::any())->method('getHostname')
+      ->willReturn($host);
+    $errors = $this->object->validate($domain);
+    static::assertNotEquals((bool) $result, (bool) count($errors));
+  }
+
+  /**
+   * @return array
+   */
+  public function provider() {
```

Comment:
Is `provider()` a standard naming convention for PHPUnit? If not, this seems like an odd name. I would think that `expectedResult()` is more clear. So it's not entirely clear how this is invoked.